manifests/rule.pp

```diff
@@ -122,47 +122,52 @@
 #     postrotate   => '/etc/init.d/nginx restart',
 #   }
 define logrotate::rule(
-                        $path            = 'UNDEFINED',
-                        $ensure          = 'present',
-                        $compress        = 'UNDEFINED',
-                        $compresscmd     = 'UNDEFINED',
-                        $compressext     = 'UNDEFINED',
-                        $compressoptions = 'UNDEFINED',
-                        $copy            = 'UNDEFINED',
-                        $copytruncate    = 'UNDEFINED',
-                        $create          = 'UNDEFINED',
-                        $create_mode     = 'UNDEFINED',
-                        $create_owner    = 'UNDEFINED',
-                        $create_group    = 'UNDEFINED',
-                        $dateext         = 'UNDEFINED',
-                        $dateformat      = 'UNDEFINED',
-                        $delaycompress   = 'UNDEFINED',
-                        $extension       = 'UNDEFINED',
-                        $ifempty         = 'UNDEFINED',
-                        $mail            = 'UNDEFINED',
-                        $mailfirst       = 'UNDEFINED',
-                        $maillast        = 'UNDEFINED',
-                        $maxage          = 'UNDEFINED',
-                        $minsize         = 'UNDEFINED',
-                        $maxsize         = 'UNDEFINED',
-                        $missingok       = 'UNDEFINED',
-                        $olddir          = 'UNDEFINED',
-                        $postrotate      = 'UNDEFINED',
-                        $prerotate       = 'UNDEFINED',
-                        $firstaction     = 'UNDEFINED',
-                        $lastaction      = 'UNDEFINED',
-                        $rotate          = 'UNDEFINED',
-                        $rotate_every    = 'UNDEFINED',
-                        $size            = 'UNDEFINED',
-                        $sharedscripts   = 'UNDEFINED',
-                        $shred           = 'UNDEFINED',
-                        $shredcycles     = 'UNDEFINED',
-                        $start           = 'UNDEFINED',
-                        $su              = 'UNDEFINED',
-                        $su_owner        = 'UNDEFINED',
-                        $su_group        = 'UNDEFINED',
-                        $uncompresscmd   = 'UNDEFINED'
-                        ) {
+  Enum['present','absent'] $ensure   = 'present',
+  Optional[
```

Comment:
There is a data type for ensurable in the module, I will use that and add some more.